### PR TITLE
PAL: Editable Display Names & Immediate Update

### PR DESCRIPTION
**Pull request by zfox23**
This PR allows users to edit their Display Name from within the PAL via an editable TextInput field. As a bonus, display names will update immediately. Other users who have their PALs open need only to refresh the PAL to see your shiny new name. Other users who open the PAL after you change your name will see the changes immediately.

I'm a little nervous about some of the changes in here because they modify code that @howard-stearns created 😄. Hopefully I didn't totally break any behavior he introduced.

**Test Plan**
This test plan requires three clients running this PR: Clients A, B, and C. Run Sandbox from Client A; Client A will have admin privileges on the domain.
* Set the Display Name to blank on all three clients.
* Using all three clients, connect to Sandbox A
* Open the PAL on all three clients.
* Depending on the order that the clients joined, each client will get one of three display names from the AvatarMixer: "Anonymous", "Anonymous_1", or "Anonymous_2". Across all three clients, verify that each client is correctly identified by their anonymous display name.
* On Client A, using the editable text field at the top of the PAL, change the display name to "Testing". (Pressing enter or de-focusing the text field updates the name.)
* On Clients B and C, refresh the PAL. Verify that Client A can now be identified as "Testing".
* On Client B, change the display name to "Testing".
* On all three clients, refresh the PAL. Verify that Client B can now be identified as "Testing_1".
* On Client A, change the display name to "Test". Then, change the display name to "Testing".
* On all three clients, refresh the PAL. Verify that Client A can now be identified as "Testing_2".
* On Client C, change the display name to "Best QA". Disconnect from Sandbox A, then reconnect to Sandbox A. On all three clients, verify that Client C can be verified as "Best QA"

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1e8effdcc42d85b6326b0c4d240088e23b36cee1 available at: https://gist.github.com/2f23d81aa696bc0e86b0f0a942010f08<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9401-1e8effd.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9401-1e8effd.dmg

File: libraries/networking/src/udt/PacketHeaders.h [hunk 1]
```diff
@@ -211,7 +211,8 @@ enum class AvatarMixerPacketVersion : PacketVersion {
     HandControllerJoints,
     HasKillAvatarReason,
     SessionDisplayName,
-    Unignore
+    Unignore,
+	ImmediateSessionDisplayNameUpdates
```
Comment (howard-stearns) [inline libraries/networking/src/udt/PacketHeaders.h]: where is this used?

Comment (howard-stearns) [inline libraries/networking/src/udt/PacketHeaders.h]: never mind. I was looking at the wrong PR!

**Comment by zfox23** [comment]
[display name obviousness](https://trello.com/c/Kq8FqEdO/73-display-name-obviousness)

File: libraries/networking/src/udt/PacketHeaders.h (hunk 1 above)
Comment (davidkelly) [inline libraries/networking/src/udt/PacketHeaders.h]: FWIW looks like there is a whitespace issue on that line.

Comment (zfox23) [inline libraries/networking/src/udt/PacketHeaders.h]: Ah yes, nice. Will fix.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 246f5fb7f5da02172c45e8c8032bb8274ffd8ef0 available at: https://gist.github.com/2f23d81aa696bc0e86b0f0a942010f08<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9401-246f5fb.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9401-246f5fb.dmg

**Comment by davidkelly** [comment]
Only QA issue:  entering very long display names didn't display them well in the edit field.  Looked good (elipsis) in the table though.

also -- if I enter a name that is 'taken', there is no indication of this (the _1 or whatever) until I refresh the PAL.  That maybe is counter-intuitive to the end users?  It makes sense -- there needs to be a round trip to the server, but I wonder if we want to address that or not.  I'm ok either way.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 59ff970d5b400867506418da8761442fdca6dc3d available at: https://gist.github.com/2f23d81aa696bc0e86b0f0a942010f08<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9401-59ff970.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9401-59ff970.dmg

**Comment by davidkelly** [comment]
Yes that fixes it.  QA PASS

**Pull request merged**